### DISCUSSION TITLE
Disabling or overwriting (some) keyboard controls

### DISCUSSION THREAD

**Discussion by RobertBoes** [vimeo/player.js#673] [General]
### Expected Behavior

I expect an option to disable keyboard controls for video's when the video is **not** set as background video.

### Actual Behavior

I have a custom full screen page, which contains a Vimeo video, when the user started playing and presses F, the video goes to fullscreen. Instead, my own element which contains the video should go to fullscreen.

### Steps to Reproduce

I have made codepen to demonstrate what I'm trying to achieve: https://codepen.io/RobertBoes/pen/NWqjmYy

Here you press on F to make the "main div" fullscreen. However, when the user plays the video (and thus has focus on the iframe/video, only the Vimeo iframe goes to fullscreen. I want to prevent this behaviour, or the ability to hook into an event.

----

Something like this would be perfect:
```
player.on('fullscreen-change', function () {
  // Do your custom fullscreen, instead of how Vimeo does this
});
```
This way, when the event is registered, Vimeo should not do anything when "F" is pressed, but instead that's handled by the third party.

Another option would be to just disable keyboard controls

**Comment by Archiefkast** [comment]
The possibility to disable keyboard controls is also required for WCAG compliance: https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts

**Comment by bskibinski** [comment]
We are now unable to embed the vimeo player on government websites because of the WCAG compliance as stated by @Archiefkast. It would be nice if we could add an URL parameter to the URL (just like youtube video's allow: disablekb) to disable keyboard controls. youtube's parameters: https://developers.google.com/youtube/player_parameters

**Comment by rianrietveld** [comment]
Hi, has there been work done on an option to disable keyboard shortcuts in Vimeo players? Like the option YouTube offers with the parameter `disablekb=1`? 

As I understand the player aims to [comply with WCAG 2.1 AA](https://vimeo.com/blog/post/accessibility-updates-to-the-vimeo-player/), then this is something that has to be addressed. 
I'd love to use the Vimeo player for websites that needs to comply with WCAG.

**Comment by germanbv** [comment]
so, this discussion has been open for a while and there is still no solution for this? this issue is a reason to stop using Vimeo premium and migrate to another service.

**Reply by julezrulez** [reply]
That's what is happening right now with clients of mine who needs to comply to WCAG 2.1 level AA. Several are leaving Vimeo.

**Comment by germanbv** [comment]
so, I've fixed this by setting: pointer-events to none for the Vimeo iframe in the CSS, for example:

div#externalVPlayer iframe 
{
	pointer-events: none;
}

with that, the users will not be able to interact with the Vimeo iframe and you can create your own onkeydown listeners.

**Comment by rianrietveld** [comment]
@bskibinski opened an issue about this, please give your support there too: https://github.com/vimeo/player.js/issues/732

**Comment by lssamlyon** [comment]
We are glad to say that the issue is now resolved with version 2.16.0 of player.js, which can now take in an optional keyboard option. If keyboard is set to false or 0, it will disable keyboard input on the player. Documentation on adding the embed option can be found here: https://github.com/vimeo/player.js#embed-options
